src/popup/modules/UserInterface.js

```diff
@@ -109,8 +109,15 @@ const refreshQrCode = throttle(() => {
         CommonMessages.hideError();
     }
 
-    QrCreator.setTextInternal(text);
-    QrCreator.generate();
+    try {
+        QrCreator.setTextInternal(text);
+        QrCreator.generate();
+    } catch (e) {
+        if (e === "Data too long") {
```

Comment:
Please add some comment explaining where that string comes from etc. (similar to how you do in the PR description, but in short obviously)